cpp/oneapi/dal/detail/common.hpp

```diff
@@ -196,6 +196,42 @@ inline constexpr std::int64_t get_data_type_size(data_type t) {
     }
 }
 
+inline constexpr std::int64_t get_data_type_align(data_type t) {
+    if (t == data_type::int8) {
+        return alignof(std::int8_t);
+    }
+    else if (t == data_type::int16) {
+        return alignof(std::int16_t);
+    }
+    else if (t == data_type::int32) {
+        return alignof(std::int32_t);
+    }
+    else if (t == data_type::int64) {
+        return alignof(std::int64_t);
+    }
+    else if (t == data_type::uint8) {
+        return alignof(std::uint8_t);
+    }
+    else if (t == data_type::uint16) {
+        return alignof(std::uint16_t);
+    }
+    else if (t == data_type::uint32) {
+        return alignof(std::uint32_t);
+    }
+    else if (t == data_type::uint64) {
+        return alignof(std::uint64_t);
+    }
+    else if (t == data_type::float32) {
+        return alignof(float);
+    }
+    else if (t == data_type::float64) {
+        return alignof(double);
+    }
+    else {
+        throw unimplemented{ dal::detail::error_messages::unsupported_data_type() };
+    }
```

Comment:
`switch` can look better here.

Reply:
Also, as this is used for aligning arrays, maybe it would be better to do: `return alignof(float *);` ?
And, isn't it also have to take into consideration the ISA? To make it more SIMD-friendly.

Reply:
else-if structure is inherited.